agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.h

```diff
@@ -47,6 +47,22 @@ class backhaul_manager : public btl::transport_socket_thread {
 
     // For agent_ucc_listener
     const std::string &get_controller_bridge_mac() { return controller_bridge_mac; }
+    /**
+     * @brief get radio mac (ruid) of registered slave based on frequency type
+     * 
+     * @param freq radio frequency to search
+     * @return radio mac of the found slave if found, otherwise empty string
+     */
+    const std::string freq_to_radio_mac(enum eFreqType freq) const;
```

Comment:
I really don't like this approach that is taken in master::db of using a MAC address as an identifier. We're doing C++ here, not tcl, so not everything is a string - we have actual *objects* to manipulate. In other words, you can just return a `std::shared_ptr<sRadioInfo>`.

Reply:
I think its easier to use the radio_mac instead of the sRadioInfo.